src/PhpDoc/TypeNodeResolver.php

```diff
@@ -333,11 +334,15 @@ private function resolveIdentifierTypeNode(IdentifierTypeNode $typeNode, NameSco
 				return new NeverType(true);
 
 			case 'list':
-				return new ArrayType(new IntegerType(), new MixedType());
+				return TypeCombinator::intersect(
```

Comment:
Because there are gonna be bugs and because this change might be pretty disruptive, I'd like if we interpreted "list" as list type only with bleeding edge.